web/src/components/MetricRatio.tsx

```diff
@@ -1,6 +1,14 @@
-export function MetricRatio({ value, total, format }) {
-  const formattedValue = Number.isFinite(value) ? format(value) : '?';
-  const formattedTotal = Number.isFinite(total) ? format(total) : '?';
+import type { Maybe } from 'types';
+
+interface MetricRatioProps {
+  value: number;
+  total: Maybe<number>;
+  format: (value: number) => string | number;
+}
+
+export function MetricRatio({ value, total, format }: MetricRatioProps) {
+  const formattedValue = Number.isFinite(value) && value ? format(value) : '?';
+  const formattedTotal = Number.isFinite(total) && total ? format(total) : '?';
```

Comment:
Why is the added check here needed? Shouldn't isFinite only return true if its a defined number?

Reply:
Unfortunately not :( 

https://stackoverflow.com/questions/66703917/why-number-isfinite-dont-have-type-guard

Reply:
Well that's weird... But that answers that question.